xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -156,6 +157,32 @@ bool CRendererMediaCodecSurface::RenderUpdateVideoHook(bool clear, DWORD flags,
     mci->RenderUpdate(srcRect, dstRect);
   }
 
+  /** @todo This is a really bad hack and not meant to be
+  * a solution. This needs proper fixing when android maintainer
+  * is back. Real solution: Block depending on next vblank,
+  * which is estimated here by assuming 2 buffers and frametime.
+  */
+  uint64_t sleep_time = 10; // 10 ms by default
+  double fps = g_graphicsContext.GetFPS();
+
+  if (fps > 0.0)
+  {
+    uint64_t now = XbmcThreads::SystemClockMillis();
```

Comment:
this is not correct. SystemClockMillis is of type unsigned int and can wrap. If you don't take this into account, you may sleep here for a very long time.